WooCommerce/Classes/ViewRelated/Orders/Order Details/Shipping Labels/WooShipping Create Shipping Labels/WooShipping Split Shipments/CollapsibleShipmentCard.swift

```diff
@@ -0,0 +1,72 @@
+import Yosemite
+import SwiftUI
+
+/// Displays a single collapsible shipment item row or grouped parent and child shipment item rows
+struct CollapsibleShipmentCard: View {
+    @State private var isCollapsed: Bool = true
+
+    private let viewModel: CollapsibleShipmentCardViewModel
+
+    init(viewModel: CollapsibleShipmentCardViewModel) {
+        self.viewModel = viewModel
+    }
+
+    var body: some View {
+        VStack(alignment: .leading, spacing: 0) {
+            mainShipmentRow
+                .padding(.horizontal, Layout.horizontalPadding)
+                .padding(.vertical, Layout.verticalPadding)
+                .background(Color(.listForeground(modal: false)))
+
+            if !isCollapsed {
+                VStack(spacing: 0) {
+                    ForEach(viewModel.childShipmentRows) { item in
+                        VStack(spacing: 0) {
+                            Divider()
+
+                            SelectableShipmentRow(viewModel: item)
+                                .padding(.leading, Layout.horizontalPadding * 2)
+                                .padding(.trailing, Layout.horizontalPadding)
+                                .padding(.vertical, Layout.verticalPadding)
+                        }
+                    }
+                }
+            }
+        }
+        .frame(maxWidth: .infinity, alignment: .center)
+        .roundedBorder(cornerRadius: Layout.borderCornerRadius, lineColor: Color(.separator), lineWidth: Layout.borderWidth)
+    }
+}
+
+private extension CollapsibleShipmentCard {
+    @ViewBuilder
+    var mainShipmentRow: some View {
+        if viewModel.childShipmentRows.isEmpty {
+            SelectableShipmentRow(viewModel: viewModel.mainShipmentRow)
+        } else {
+            Button(action: {
+                withAnimation {
+                    isCollapsed.toggle()
+                }
+            }, label: {
+                ZStack(alignment: .topTrailing) {
+                    SelectableShipmentRow(viewModel: viewModel.mainShipmentRow)
```

Comment:
Do we need to allow toggling when tapping on any white space in the row? We might need the `contentShape` modifier for that. Currently, the row is only toggled when tapping either on the chevron icon or on the image/labels.

Reply:
Nice catch. Thanks for the suggestion. Applied it in https://github.com/woocommerce/woocommerce-ios/pull/15397/commits/f3a3f7e53bf7e4a30b79ba7f5f3d5ca7c8a8a8b4